File: python/nwd.py
def nwd(a, b):
    while b != 0 :
        c = a % b
        a, b = b, c

    return abs(a)


def main():
    a = int(input('Podaj pierwszą liczbę: '))
    b = int(input('Podaj drugą liczbę: '))

    print(f"NWD: { nwd(a, b) }")


if __name__ == '__main__':
    main()

File: c/nwd.c
#include <stdio.h>

int nwd(int a, int b) {
    int c;
    while (b != 0) {
        c = a % b;
        a = b;
        b = c;
    }

    return (a < 0) ? -a : a; 
}

int main() {
    int a, b;

    printf("Podaj pierwszą liczbę: ");
    scanf("%d" ,&a);
    printf("Podaj drugą liczbę: ");
    scanf("%d" ,&b);

    printf("NWD: %d\n", nwd(a, b));
}